internal/rules/resources_test.go

```diff
@@ -0,0 +1,64 @@
+package rules_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/titpetric/factory"
+
+	"github.com/crusttech/crust/internal/auth"
+	"github.com/crusttech/crust/internal/rules"
+	"github.com/crusttech/crust/system/types"
+)
+
+func TestRules(t *testing.T) {
+	user := &types.User{ID: 1337}
+	ctx := auth.SetIdentityToContext(context.Background(), user)
+
+	db := factory.Database.MustGet()
+	mustFail(t, db.Transaction(func() error {
+		db.Insert("sys_user", user)
+		var i uint64 = 0
+		for i < 5 {
+			db.Insert("sys_team", types.Team{ID: i, Name: fmt.Sprintf("Team %d", i)})
+			i++
+		}
+		db.Insert("sys_team_member", types.TeamMember{TeamID: 1, UserID: user.ID})
+		db.Insert("sys_team_member", types.TeamMember{TeamID: 2, UserID: user.ID})
+
+		resources := rules.NewResources(ctx, db)
+
+		// default (unset=deny)
+		{
+			mustFail(t, resources.CheckAccess("channel:1", "edit"))
```

Comment:
why not split this into resource type & resource id?

Reply:
Mostly to have a unique resource id that identifies the type as well as the record (it comes from daasi api-s as well so we need to be somewhat compatible here). As for how the record ID is put together, there's a common function available for record entries under sam/types currently (ResourceID() or something like that).